add synopsys creg gpio driver support for arc hsdk and em_starterkit boards

IRISZZW opened the pull request:
Add single-register MMIO GPIO driver for complex cases where only several fields in register belong to
GPIO lines and each GPIO line owns a field with different length and on/off value.

Such CREG GPIOs are used in Synopsys em_starterkit and HSDK boards for SPI CS pin control.

fix: #35377

IRISZZW committed:
boards: arc: hsdk: add creg_gpio driver support

Add creg_gpio driver support for hsdk board.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>

IRISZZW committed:
boards: arc: em_starterkit: add creg_gpio driver support

Add creg_gpio driver support for em_starterkit board.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>

mbolivar-nordic reviewed (approved):
DT bits seem reasonable

IRISZZW committed:
driver: gpio: add initial support for synopsys creg gpio

Add single-register MMIO GPIO driver for complex cases where
only several fields in register belong to GPIO lines and each GPIO
line owns a field with different length and on/off value.

Such CREG GPIOs are used in Synopsys em_starterkit and HSDK boards.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>
